src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -145,6 +145,8 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     stateParams.messageCounterManager = chip::Platform::New<secure_channel::MessageCounterManager>();
 
     ReturnErrorOnFailure(stateParams.fabricTable->Init(params.fabricIndependentStorage));
+    ReturnErrorOnFailure(
+        stateParams.fabricTable->AddFabricDelegate(chip::Platform::New<ControllerFabricDelegate>(stateParams.sessionMgr)));
```

Comment:
I think it would be safer to do:
```suggestion
    auto delegate = chip::Platform::MakeUnique<ControllerFabricDelegate>(stateParams.sessionMgr);
    ReturnErrorOnFailure(stateParams.fabricTable->AddFabricDelegate(delegate.get()));
    delegate.release();
```
so we're not depending on exactly when AddFabricDelegate does or does not fail.

Reply:
Fixed